### PR TITLE
VPlay: honour the pixel aspect ratio

### PR DESCRIPTION
**Pull request by gelisam**
Some video cameras produce video files with non-square pixels, so when playing on a screen with square pixels, video players need to compensate by stretching the image.

### CODE REVIEW

File: demo/VPlay.hs [hunk 1]
```diff
@@ -266,13 +266,28 @@ videoPlayer cfg src = do
...
+      textureHeight <- liftIO $ getHeight ctx
+
+      -- Compute window size. If the pixels aren't square, stretch the window,
+      -- SDL will automatically scale the texture to fit.
+      par <- liftIO $ getAspectRatio ctx
+
+      let rawAspectRatio :: Double
+          rawAspectRatio = fromIntegral (numerator par) / fromIntegral (denomenator par)
+
+          -- 0 means "unknown", so we default to 1, meaning "square pixels"
```
[the remark this answers is not in the document]
Comment (gelisam) [reply demo/VPlay.hs]: Done.

**Comment by acowley** [comment]
I think I may have messed you up by merging the ticks per frame PR first. Can you make sure this merges cleanly?

As usual, thank you for identifying and fixing this issue.

**Comment by gelisam** [comment]
> I think I may have messed you up by merging the ticks per frame PR first.

Nah, the two PRs were touching the same lines and so I knew that whichever one you merged first would cause conflicts with the other one.

**Comment by gelisam** [comment]
Thanks for the commit bit. Would you still like me to send future improvements as PRs, or should I commit them directly?

**Comment by acowley** [comment]
Use your judgment. I think PRs are nice so people can ask questions and catch goofs, but moving things along is also valuable.

If it’s a new feature, I’d like to let a PR sit for a day in case anyone’s available to take a look.

**Pull request merged**